elastic/beats
# [Metricbeat] allow partial region and zone in googlecloud module config

## What does this PR do?

This PR is to enable partial region and partial zone in googlecloud module config.  User can choose to use wildcard `*` or not. For example:
```
- module: googlecloud
  metricsets:
    - compute
  region: "us-west"
  project_id: "elastic-observability"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 60s
```
This config will enable data collection from all regions start with `us-west`, which includes `us-west1`, `us-west2`, `us-west3` and `us-west4`.
 
```
- module: googlecloud
  metricsets:
    - compute
  region: "us-*"
  project_id: "elastic-observability"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 60s
```
Similarly this config will enable data collection from all regions start with `us-`, which includes all regions starts with `us-west`, `us-east` and `us-central`.

## Why is it important?

This will allow users to not copy paste region/zone names when they want to monitor several regions/zones. If user wants to monitor all regions in both US and Europe, they still need to copy paste but that's very minimum. For example:
```
- module: googlecloud
  metricsets:
    - compute
  region: "us-*"
  project_id: "elastic-observability"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 60s
- module: googlecloud
  metricsets:
    - compute
  region: "europe-*"
  project_id: "elastic-observability"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 60s
```

## Checklist




## How to test this PR locally

There are two parts need to be tested. One for regions and another for zones.

- Test for regions:
Enable googlecloud with this config and you should see metrics collection from all regions start with `us-`.
```
- module: googlecloud
  metricsets:
    - compute
  region: "us-"
  project_id: "elastic-observability"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 60s
```

- Test for zones:
Enable googlecloud with this config and you should see metrics collection from all zones start with `europe-west3-`, which includes `europe-west3-a`, `europe-west3-b` and `europe-west3-c`.
```
- module: googlecloud
  metricsets:
    - compute
  zone: "europe-west3-"
  project_id: "elastic-observability"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 60s
```

## Related issues

- Relates https://github.com/elastic/beats/issues/17835
